sklearn/ensemble/partial_dependence.py

```diff
@@ -306,8 +306,8 @@ def convert_feature(fx):
                 l.append(feature_names[i])
             names.append(l)
     except IndexError:
-        raise ValueError('features[i] must be in [0, n_features) '
-                         'but was %d' % i)
+        raise ValueError('features[i] must be in [0, len(feature_names)={0} ) '
+                         'but was {1}'.format(len(feature_names), i))
```

Comment:
I find it a bit odd that you specify that "features[i] must be in a range", but then show the user the value of "i" in the error message.